packages/web/hooks/use-swap.tsx

```diff
@@ -371,11 +379,44 @@ export function useSwap(
     return balance.toDec().lt(amountWithSlippage);
   }, [inAmountInput.balance, inAmountInput.amount, maxSlippage, quoteType]);
 
+  /**
+   * Create refs so we can wait for them to finish in the sendTradeTokenInTx function
+   * This is a safety measure to prevent signed swaps from failing due to stale gas amount
+   * e.g. after enabling 1CT session the gas estimation needs to be updated before sending the swap
+   */
+  const isLoadingNetworkFeeRef = useRef(isLoadingNetworkFee);
+  const networkFeeErrorRef = useRef(networkFeeError);
+  useEffect(() => {
+    networkFeeErrorRef.current = networkFeeError;
+    isLoadingNetworkFeeRef.current = isLoadingNetworkFee;
+  }, [isLoadingNetworkFee, networkFeeError]);
+
   /** Send trade token in transaction. */
   const sendTradeTokenInTx = useCallback(
     () =>
       new Promise<"multiroute" | "multihop" | "exact-in">(
         async (resolve, reject) => {
+          // Wait for network fee to load, retry up to 10 times
+          let retries = 0;
+          while (isLoadingNetworkFeeRef.current && retries < 10) {
+            if (networkFeeErrorRef.current) {
+              return reject(new Error(networkFeeErrorRef.current.message));
+            }
+            await new Promise((resolve) => setTimeout(resolve, 100));
+            retries++;
+          }
+
+          // After retries, check if still loading
+          if (isLoadingNetworkFeeRef.current) {
+            return reject(
+              new Error("Network fee is still loading after 1 second")
+            );
+          }
+
+          if (networkFeeErrorRef.current) {
+            return reject(new Error(networkFeeErrorRef.current.message));
+          }
```

Comment:
This works, but it feels a bit hacky. I wonder if we could use a global state with Zustand to check if the review modal’s 1CT is enabled? Then, we could construct a multi-message within the swap hooks. I have a hunch this approach would be more accurate and require less code. What do you think?

Reply:
> This works, but it feels a bit hacky. 

Indeed :) I don't like it either. 

>  I wonder if we could use a global state with Zustand to check if the review modal’s 1CT is enabled? Then, we could construct a multi-message within the swap hooks.

This could work in the normal user flow and allow the user to `Continue Anyway` with the swap, even if the 1CT session has an overspend limit error. I like this idea a lot, but if we were to do it, I would prefer to tackle it in another PR. 

TL;DR;
This is me being extra cautious to prevent swap tx failure after enabling 1CT through the review order flow. 
It is rarely necessary, and we could revert the whole change if it's too much/too hacky.  

Full context: 

Using 1CT needs more gas, and the swap tx would fail with the old gas estimate after 1CT is enabled.   

When 1CT is enabled through the review order flow, the useSwap hook needs to update its internal state while the `createSession.mutate` and its `onSuccess` callback are running.

We wait for the session to update in the `onSuccess,` but the network fee invalidation in `useSwap` starts after the session update  (ecd25aa9ab83). 

In the `onSuccess`,  we don't have any info about when the `useSwap` will be ready. 

So we either
1. Take the "readiness signal" from `SwapTool > swapState` -> `ReviewOrder` -> `useOneClickTradingSessionManager` 
2. Wait for an arbitrary amount of time and hope that gas estimation will be ready after that. 
3. Make `SwapTool > sendSwapTx` smart enough to wait 
4. Make the `useSwap > sendTradeTokenInTx` smart enough to wait 
5. Ditch the safety mechanism 

I chose 3. to implement this protection with the rest of the other checks to ensure that the swap tx can be executed. 

What do you think would be a good next step?

Reply:
imo we should keep this as is and address this in a separate PR